submap_service: "/elevation_mapping_long_range/get_submap"
elevation_map_topic: "/image_to_gridmap_demo/grid_map"
robot_frame_id: "base"
map_frame_id: "map"
min_update_rate: 0.0
map_center_x: 1.0
map_center_y: 0.0
map_length_x: 6.0
map_length_y: 6.0
traversability_default: 0.5

traversability_map_filters:
 - name: surfaceNormalsFilter
   type: traversabilityFilters/SurfaceNormalsFilter
   params:
    estimation_radius: 0.1
    surface_normal_positive_axis: "z"
 - name: slopeFilter
   type: traversabilityFilters/SlopeFilter
   params:
    map_type: traversability_slope
    critical_value: 0.610865
 - name: stepFilter
   type: traversabilityFilters/StepFilter
   params:
    map_type: traversability_step
    critical_value: 0.12
    first_window_radius: 0.08
    second_window_radius: 0.08
    critical_cell_number: 5
 - name: roughnessFilter
   type: traversabilityFilters/RoughnessFilter
   params:
    map_type: traversability_roughness
    critical_value: 0.02
    estimation_radius: 0.05
 - name: weightedSumFilter
   type: gridMapFilters/WeightedSumFilter
   params:
    layers: [traversability_slope, traversability_step, traversability_roughness]
    weights: [2, 2, 1]
    layer_out: traversability
    normalize: 1
# - name: minFilter
#   type: gridMapFilters/MinFilter
#   params:
#    filter_types: [traversability_slope, traversability_step, traversability_roughness]
#    type_out: traversability
#  - name: thresholdFilter
#    type: gridMapFilters/ThresholdFilter
#    params:
#     lower_threshold: 0.2
##     upper_threshold: 0.9
#     set_to: 0.0
#     threshold_types: [traversability]
#  - name: thresholdFilter
#    type: gridMapFilters/ThresholdFilter
#    params:
##     lower_threshold: 0.2
#     upper_threshold: 0.9
#     set_to: 1.0
#     threshold_types: [traversability]
#  - name: deletionFilter
#    type: gridMapFilters/DeletionFilter
#    params:
#     deletion_types: [traversability_slope, traversability_step, traversability_roughness]
 - name: deletionFilter
   type: gridMapFilters/DeletionFilter
   params:
    layers: [variance, surface_normal_x, surface_normal_y, surface_normal_z]
